oauthproxy.go

```diff
@@ -810,10 +858,43 @@ func (p *OAuthProxy) redeemCode(req *http.Request) (*sessionsapi.SessionState, e
 	if s.ExpiresOn == nil {
 		s.ExpiresIn(p.CookieOptions.Expire)
 	}
+	s.Authenticated = true
 
 	return s, nil
 }
 
+func generateCodeChallenge(method, codeVerifier string) (string, error) {
```

Comment:
Related to my previous note on the random function -- I think this logic also might be best placed near our other random/encryption helpers under `pkg/encryption`.